### PR TITLE
ci: move the CI to GitHub Actions

### PR DESCRIPTION
**Pull request by mrc0mmand**

### CODE REVIEW

**Comment by mrc0mmand** [comment]
@keszybz this should be almost ready to go. A couple of questions:

1) Which python versions should we cover (for now)
2) The tests currently fail with python 2.7 - should they get fixed or we don't care about 2.7 anymore?

**Comment by mrc0mmand** [comment]
>     2. The tests currently fail with python 2.7 - should they get fixed or we don't care about 2.7 anymore?

I see, the first test was easy enough to fix, but in the second case the original change actually broke python 2.7 compatibility. i.e. since 4c9a2419 the code doesn't work on 2.7 anymore, since `datetime.astimezone()` requires argument on 2.7, whereas on 3.x it defaults to the local timezone.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python }}
+          architecture: x64
+
+      - name: Install dependencies
+        run: |
+          sudo apt -y update
+          sudo apt -y install gcc libsystemd-dev
+          pip install pytest sphinx
```
Comment (behrmann) [inline .github/workflows/build.yml:41]: This leaves a bit open which pip this is. I think the general recommendation is to go through `python -m pip` to get the exact pip for the python one is using. In that sense might this make more sense in the next step?

**Review by behrmann — COMMENTED** [review]
Great work. This would be a good step for #103.

File: .github/workflows/build.yml (hunk 1 above)
Comment (mrc0mmand) [reply .github/workflows/build.yml]: Good point, updated, thanks!

**Review by jamacku — COMMENTED** [review]
Would it make sense to run these workflows on push events as well?
Maybe adding Dependabot would make sense as well, to keep dependencies up to date?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      group: ${{ github.workflow }}-${{ matrix.python }}-${{ github.ref }}
+      cancel-in-progress: true
+    strategy:
+      fail-fast: false
+      matrix:
+        python: [
+          "3.7",
+          "3.8",
+          "3.9",
+          "3.10",
```
Comment (keszybz) [inline .github/workflows/build.yml:24]: We'll need to add 3.11 here…

**Review by keszybz — COMMENTED** [review]
Thanks, looks great.

**Comment by mrc0mmand** [comment]
> Would it make sense to run these workflows on push events as well? Maybe adding Dependabot would make sense as well, to keep dependencies up to date?

That's definitely on the TODO list.

File: .github/workflows/build.yml (hunk 2 above)
Comment (mrc0mmand) [reply .github/workflows/build.yml:24]: Hehe, yeah, it's been a while.

**Comment by keszybz** [comment]
> > 2. The tests currently fail with python 2.7 - should they get fixed or we don't care about 2.7 
> 
> I see, the first test was easy enough to fix, but in the second case the original change actually broke python 2.7 compatibility. i.e. since [4c9a241](https://github.com/systemd/python-systemd/commit/4c9a241949067fc8d55f3ea12170ad364bd8b18d) the code doesn't work on 2.7 anymore, since `datetime.astimezone()` requires argument on 2.7, whereas on 3.x it defaults to the local timezone. The rationale is even mentioned in the commit message, so it looks like we indeed don't care about 2.7 anymore :-)

I don't want to spend time on 2.7 myself, but if somebody comes up with a patch, I guess we'd merge it. If support for 2.7 is fully removed, then we should also drop the C parts, where actually there's a lot of code duplication for the old C API.

**Pull request merged**